### PR TITLE
Factory migration, bump dependencies and artifacts

### PR DESCRIPTION
**Pull request by jamesreprise**
## What is the goal of this PR?

We've migrated from grabl to Factory.

## What are the changes implemented in this PR?

We've:
* reworded instances of "grabl" to "factory"
* changed 16-core-32-gb machines to 8-core-32-gb machines (as GCP doesn't offer 16-core-32-gb machines in the N series and we deem the memory usage characteristics to be more important)
* bumped dependencies
* bumped artifacts

### CODE REVIEW

**Comment by jamesreprise** [comment]
We're blocked on fixes to Cluster and Core, once released we can bump these artifacts and merge when CI passes.

**Comment by jamesreprise** [comment]
Ready for review and merge once CI passes!

File: dependencies/vaticle/artifacts.bzl [hunk 1]
```diff
@@ -25,7 +25,7 @@ def vaticle_typedb_artifacts():
         artifact_name = "typedb-server-{platform}-{version}.{ext}",
         tag_source = deployment["artifact.release"],
         commit_source = deployment["artifact.snapshot"],
-        commit = "322787dbea6d91718543573f2bfed4c9c008aea9",
```
Comment (flyingsilverfin) [inline dependencies/vaticle/artifacts.bzl:28]: why not use `tag = 2.13.0` for both @jamesreprise ?

File: dependencies/vaticle/repositories.bzl [hunk 2]
```diff
@@ -21,19 +21,19 @@ def vaticle_dependencies():
...
-        commit = "a8b3b714a5e0562d41f4c05ca8f266d48b7d0fd3",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
+        commit = "5be6d949ca1e04e4179ea6acb3432be713b9dfb8",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
     )
 
 def vaticle_typedb_client_java():
     git_repository(
         name = "vaticle_typedb_client_java",
         remote = "https://github.com/vaticle/typedb-client-java",
-        commit = "5717b9cb0c63e0e1c4457ad8e5ad5141d3f04556",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_typedb_client_java
+        commit = "73754389f6c68fd1bbd10953d5441336d379b0b0",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_typedb_client_java
```
Comment (flyingsilverfin) [inline dependencies/vaticle/repositories.bzl]: use released `tag = 2.12.0`

File: dependencies/vaticle/repositories.bzl [hunk 3]
```diff
@@ -21,19 +21,19 @@ def vaticle_dependencies():
...
     )
 
 def vaticle_factory_tracing():
     git_repository(
         name = "vaticle_factory_tracing",
         remote = "https://github.com/vaticle/factory-tracing",
-        commit = "a4d5e36b1b0c56bd370e5a7583d628de9272f086"  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_factory_tracing
+        commit = "fe732088df298732216fb8190d6667bf94812057"  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_factory_tracing
```
Comment (flyingsilverfin) [inline dependencies/vaticle/repositories.bzl]: use released `tag = 2.12.0`

**Review by flyingsilverfin — APPROVED** [review]
Just bump to the released versions, then feel free to merge!

**Pull request merged**